cmd/govim/main.go

```diff
@@ -428,3 +435,29 @@ func (g *govimplugin) Shutdown() error {
 	}
 	return nil
 }
+
+func (g *govimplugin) defineHighlights() {
+	warnColor := 166 // Orange
+	if vimColors, err := strconv.Atoi(g.ParseString(g.ChannelExpr(`&t_Co`))); err != nil || vimColors < 256 {
+		warnColor = 3 // Yellow, fallback when the terminal doesn't support at least 256 colors
+	}
+	g.ChannelExf("highlight default %s term=underline cterm=underline gui=undercurl ctermfg=1 guisp=Red", types.HighlightErr)
+	g.ChannelExf("highlight default %s term=underline cterm=underline gui=undercurl ctermfg=%d guisp=Orange", types.HighlightWarn, warnColor)
+	g.ChannelExf("highlight default %s term=underline cterm=underline gui=undercurl ctermfg=6 guisp=Cyan", types.HighlightInfo)
+	g.ChannelExf("highlight default link %s %s", types.HighlightHint, types.HighlightInfo)
+
+	g.ChannelExf("highlight default link %s ErrorMsg", types.HighlightSignErr)
+	g.ChannelExf("highlight default %s ctermfg=15 ctermbg=%d guisp=Orange guifg=Orange", types.HighlightSignWarn, warnColor)
+	g.ChannelExf("highlight default %s ctermfg=15 ctermbg=6 guisp=Cyan guifg=Cyan", types.HighlightSignInfo)
+	g.ChannelExf("highlight default link %s %s", types.HighlightSignHint, types.HighlightSignInfo)
+}
+
+func (g *govimplugin) highlightDiagnostics() bool {
+	if g.Flavor() != govim.FlavorVim && g.Flavor() != govim.FlavorGvim {
+		return false
+	}
+	if os.Getenv(testsetup.EnvDisableHighlightDiags) == "true" {
```

Comment:
With a normal config value this method can, I think, go?

Reply:
You will however need to add to the logic in

```
func (v *vimstate) setConfig(args ...json.RawMessage) (interface{}, error)
```

Reply:
indeed, removed and added config instead